src/components/CompareResults/beta/DownloadButton.tsx

```diff
@@ -14,9 +15,21 @@ const styles = {
 };
 
 function DownloadButton() {
+  const results = useAppSelector((state) => {
+    const { data } = state.compareResults;
+    return data;
+  });
+
   return (
     <div className={styles.downloadButton}>
-      <Button disabled>Download JSON</Button>
+      <Button
+        href={`data:text/json;charset=utf-8,${encodeURIComponent(
+          JSON.stringify(results),
+        )}`}
+        download='perf-compare.json'
+      >
+        Download JSON
```

Comment:
@beatrice-acasandrei  Here I have not changed the format of results from the redux store. I am not sure if this would be correct or not. Do I need to convert the results into a specific format? 
(Results are converted to JSON as it is here)